keanu-project/src/test/java/io/improbable/keanu/util/ProgressBarTest.java

```diff
@@ -0,0 +1,95 @@
+package io.improbable.keanu.util;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ProgressBarTest {
+
+    private AtomicReference<Runnable> progressUpdateCall;
+    private ProgressBar progressBar;
+    private ByteArrayOutputStream byteArrayOutputStream;
+
+    @Before
+    public void setup() throws UnsupportedEncodingException {
+
+        byteArrayOutputStream = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream(byteArrayOutputStream, true, "UTF-8");
+        ScheduledExecutorService scheduler = mock(ScheduledExecutorService.class);
+
+        progressUpdateCall = new AtomicReference<>(null);
+
+        when(scheduler.scheduleAtFixedRate(any(), anyLong(), anyLong(), any()))
```

Comment:
This allows you to avoid the creation of a new thread. Making assertion on a multithreaded test unit is hard but if I mock the scheduler then I can mock out various timings of calls to the unit under test and when the scheduler gets triggered.